docs/avro.rst

```diff
@@ -133,7 +100,15 @@ field, which is missing in the old data.
 Forward Compatibility
 ^^^^^^^^^^^^^^^^^^^^^
 
-Forward compatibility means that data encoded with a newer schema can be read with an older schema.
+``FORWARD`` compatibility means that data produced with a new schema can be read by consumers using the last schema, even though they may not be able to use the full capabilities of the new schema.
+For example, if there are three schemas for a subject that change in order `X-2`, `X-1`, and `X` then ``FORWARD`` compatibility ensures that data written by producers using the new schema `X` can be processed by consumers using schema `X` or `X-1`, but not necessarily `X-2`.
+If data produced with a new schema needs to be read by consumers using all registered schemas, not just the last two schemas, then use ``FORWARD_TRANSITIVE`` instead of ``FORWARD``.
+For example, if there are three schemas for a subject that change in order `X-2`, `X-1`, and `X` then ``FORWARD_TRANSITIVE`` compatibility ensures that data written by producers using the new schema `X` can be processed by consumers using schema `X`, `X-1`, or `X-2`.
+
+* ``FORWARD``: data produced using schema `X` can be ready by consumers with schema `X` or `X-1`
+* ``FORWARD_TRANSITIVE``: data produced using schema `X` can be ready by consumers with schema `X`, `X-1`, or `X`
+
+An example of a forward compatible schema modification is adding a new field. In most data formats, consumers that were written to process events without the new field will be able to continue doing so even when they receive new events that contain the new field.
```

Comment:
I would also mention that there are no guarantees that consumers that are using the new schema (and therefore expect that the new field will exist) will be able to process events with the old schema (where the new field is missing).
This basically means that you need to upgrade all producers and make sure old data isn't available for consumers before upgrading a single consumer.

Reply:
Ack